### PR TITLE
Fix digestFunction in grpc remote downloader

### PR DESCRIPTION
**Pull request by hunshcn**
fix https://github.com/bazelbuild/bazel/issues/22863

### CODE REVIEW

**Comment by hunshcn** [comment]
I'm new to java and this repo. I didn't know how to update BUILD file and run format. And no development doc.

Can someone tell me how to do it?

**Comment by fmeum** [comment]
Formatting is done with https://github.com/google/google-java-format. There is no automation for BUILD file maintenance such as Gazelle.

File: src/main/java/com/google/devtools/build/lib/remote/GrpcCacheClient.java [hunk 1]
```diff
@@ -329,16 +330,23 @@ public ListenableFuture<Void> uploadActionResult(
 
   @Override
   public ListenableFuture<Void> downloadBlob(
-      RemoteActionExecutionContext context, Digest digest, OutputStream out) {
+      RemoteActionExecutionContext context,
+      Digest digest,
+      @Nullable DigestFunction.Value digestFunction,
```
Comment (EdSchouten) [inline src/main/java/com/google/devtools/build/lib/remote/GrpcCacheClient.java:335]: Out of curiosity, why would we permit this to be null?

Comment (hunshcn) [reply src/main/java/com/google/devtools/build/lib/remote/GrpcCacheClient.java:335]: GrpcCacheClient implement RemoteCacheClient. For RemoteCacheClient, digestFunction could be null.

This is my first time to write java. Please point out my problems if you have any misunderstanding.

**Pull request closed without merging**